### PR TITLE
First Website Push

### PR DESCRIPTION
**Pull request by Xenolphthalein**
First push of the source code of the website.
It's written in vue.js with the quasar framework.
I'll update the README.md so that others know how to setup the dev env.

### CODE REVIEW

**Comment by Zeroto521** [comment]
we could use **GitHub Action** to compile the source codes on a specific **branch** when there are any commit.
And it also could get a good code reading for human.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    'import/namespace': 'error',
+    'import/default': 'error',
+    'import/export': 'error',
+    'import/extensions': 'off',
+    'import/no-unresolved': 'off',
+    'import/no-extraneous-dependencies': 'off',
+    'import/prefer-default-export': 'off',
+    'prefer-promise-reject-errors': 'off',
+
+
```
Comment (Zeroto521) [inline .eslintrc.js:65]: I will add some rules to handle such bugs as follows.

```
    // fix `Strings must use single quote`
    "quotes": [1, "single"],
    "quotes": [0, "double"],

    // fix `'LF' but found 'CRLF' linebreak-style`
    'linebreak-style': ["off", "windows"],
```

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    'import/named': 'error',
+    'import/namespace': 'error',
+    'import/default': 'error',
+    'import/export': 'error',
+    'import/extensions': 'off',
+    'import/no-unresolved': 'off',
+    'import/no-extraneous-dependencies': 'off',
+    'import/prefer-default-export': 'off',
+    'prefer-promise-reject-errors': 'off',
+
```
Comment (Zeroto521) [inline .eslintrc.js:64]: I will add some rules to handle such bugs.

```json
    // fix `Strings must use single quote`
    "quotes": [1, "single"],
    "quotes": [0, "double"],

    // fix `'LF' but found 'CRLF' linebreak-style`
    'linebreak-style': ["off", "windows"],
```

File: src/layouts/MainLayout.vue [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        <q-btn size="sm" flat stretch icon="mdi-github" type="a" href="https://github.com/Flow-Launcher/Flow.Launcher">
+          &nbsp;Flow-Launcher/Flow.Launcher
+        </q-btn>
+      </q-toolbar>
+      <q-toolbar class="bg-secondary shadow-8">
+        <q-tabs align="left" inline-label>
+          <q-tab icon="mdi-home" label="Start" @click="scrollToElement('home')" />
+          <q-tab icon="mdi-cogs" label="Features"  @click="scrollToElement('features')" />
+          <q-tab icon="mdi-download" label="Download"  @click="scrollToElement('download')" />
```
Comment (Zeroto521) [inline src/layouts/MainLayout.vue:22]: I think it could add a **Descriptions tab** after **Features tab**.

File: src/pages/Index.vue [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+      this.possible_results.forEach((possibleResult) => {
+        if (resultList.length < 4) {
+          if (possibleResult.title.toLowerCase().includes(this.query.toLowerCase())) {
+            resultList.push(possibleResult);
+          }
+        }
+      });
+      return resultList;
+    },
+  },
```
Comment (Zeroto521) [inline src/pages/Index.vue:244]: I like this part. It's great. It really likes a Flow in the browser.

File: src/pages/Index.vue [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+      }
+
+      const mathRegex = /^(?:\d+|\+|-|\/|\*|\(|\))*$/gm;
+      if (this.query.match(mathRegex)) {
+        const mathString = this.query.replace(/[^-()\d/*+.]/g, '');
+
+        resultList.push(
+          {
+            // eslint-disable-next-line no-eval
+            icon: '/images/calculator.png', title: eval(mathString), subtitle: 'Calculated from your expression', link: '', color: '',
```
Comment (Zeroto521) [inline src/pages/Index.vue:190]: The calculator function may need to be deleted. I can't figure out another better way to handle the `eval` function.

Just like the error says: `eval can be harmful  no-eval`

**Review by Zeroto521 — APPROVED** [review]
Ok, it is a great step for flow.

Next step we make it to a real website via GitHub Action. 
I need some time to figure out it.

Thanks for your PR.

File: src/pages/Index.vue (hunk 5 above)
Comment (Xenolphthalein) [reply src/pages/Index.vue:190]: Yeah, eval *can* be harmful, but In this context and with the string sanitizing in front of it it's harmless. Especially in our case - the website never pushes any data to a database, so the only experience someone malicious could break is his own one.

File: src/layouts/MainLayout.vue (hunk 3 above)
Comment (Xenolphthalein) [reply src/layouts/MainLayout.vue:22]: I included the Description directly under the start section. As the website is not that long, the auto scroll feature with the tabs isn't as effective as you see the content right in front of you even without scrolling. That's why I choose that the menu point Start will suffice for the description.

**Review by jjw24 — COMMENTED** [review]
💯

**Pull request merged**